app/src/main/java/com/cornellappdev/uplift/ui/components/capacityreminder/CapacityThreshold.kt

```diff
@@ -0,0 +1,163 @@
+package com.cornellappdev.uplift.ui.components.capacityreminder
+
+import android.annotation.SuppressLint
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableFloatStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.uplift.util.GRAY00
+import com.cornellappdev.uplift.util.GRAY01
+import com.cornellappdev.uplift.util.GRAY03
+import com.cornellappdev.uplift.util.GRAY04
+import com.cornellappdev.uplift.util.PRIMARY_YELLOW
+import com.cornellappdev.uplift.util.montserratFamily
+import kotlin.math.min
+
+/**
+ * Section with slider to set the capacity threshold for the capacity reminder feature.
+ *
+ * @param sliderVal the initial value of the slider
+ * @param onSliderValChange the callback when the slider value changes
+ */
+@SuppressLint("UseOfNonLambdaOffsetOverload")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CapacityThreshold(
+    sliderVal: Float = 0.5f,
+    onSliderValChange: (Float) -> Unit = {}
+) {
+    var sliderValue by remember { mutableFloatStateOf(sliderVal) }
```

Comment:
Hmm there seems to be some clashing design systems here. There are two main choices you can make when making a component:

1. (More robust; more work) Take all data in through parameters, and only expose events as anonymous function parameters. Everything is in parameters; the component performs no logic.
2. (Less robust; less work) Take in an initial value (if applicable), and change the component state INTERNALLY with a mutable state.

Here, it seems like you're doing BOTH. You're correctly taking in a new slider val and even providing the option of an `onSliderValChange`, but then also have a state internally here.

I guess to reiterate, to dispel confusion, we should "contractually" (like, CS 2110 speaking,) expect whoever's using this component to properly update `sliderVal` as `onSliderValChange` is called, such that the component actually visually updates to the right stuff. 

You can see what I mean if you try to use a `TextField` but don't actually update the `text` input, for example. You can type stuff into the keyboard, but because you fail to contractually fulfill the expectation of `onTextChanged`, the component doesn't actually visually update.

Reply:
Thanks for the detailed review and the explanation on the "contractual" design system, I removed the internal states for a couple components so that the changes are dependent on the parameters to the component. Hopefully my new commit reflects the changes you were talking about :)